client/src/context/useTour.tsx

```diff
@@ -0,0 +1,47 @@
+import { useContext, useState, createContext, PropsWithChildren, useEffect } from 'react';
+import { ReactourStep } from 'reactour';
+import ITourContext from '../interface/TourContext';
+
+export const TourContext = createContext<ITourContext>({} as ITourContext);
+
+interface TourProviderProps {
+  steps: ReactourStep[];
+}
+
+export const TourProvider = ({ steps, children }: PropsWithChildren<TourProviderProps>): JSX.Element => {
+  const [isTourMode, setIsTourMode] = useState<boolean>(false);
+  const [hasSeenTour, setHasSeenTour] = useState<boolean>(false);
+  const [tourSteps] = useState<ReactourStep[]>(steps || []);
+
+  useEffect(() => {
+    const hasSeenTour = localStorage.getItem('hasSeenTour');
```

Comment:
For this project, just utilizing localStorage works however how it works currently is pretty static to only the tour of the board. For extending this and working with showcasing other features (i.e., teams UI), the implementation would have to be touched up. However, the TourProvider is flexible with updating the steps so just some tweaking for a better tour system.